activities.el

```diff
@@ -503,6 +514,39 @@ To be called from `kill-emacs-hook'."
 
 ;;;; Functions
 
+(defun activities--map-window-state-leafs (state func)
+  "Call FUNC on all the leaf nodes in window-state STATE.
+The resulting values are returned as a list."
+  (let (ret)
+    (cl-labels ((map-leafs (state func)
+		  (pcase state
+		    (`(leaf . ,_attrs)
+		     (push (funcall func state) ret))
+		    ((pred proper-list-p)
+		     (if-let ((leaf-pos (cl-position 'leaf state)))
+			 (push (funcall func (cl-subseq state leaf-pos)) ret)
+		       (dolist (s state) (map-leafs s func)))))))
+      (map-leafs state func))
+    (nreverse ret)))
+
+(defun activities--buffers-and-files (state)
+  "Return a list of buffers and files from STATE.
+STATE is a window-state.  The returned list contains elements of
+form (BUFFER . FILE) assocaited with the activity."
+  (activities--map-window-state-leafs
+   (activities-activity-state-window-state state)
+   (lambda (leaf)
+     (let ((rec (map-nested-elt (cdr leaf) '(parameters activities-buffer))))
+       (cons (activities-buffer-name rec) (activities-buffer-filename rec))))))
+
+(defun activities--buffers-and-files-differ-p (bf1 bf2)
+  "Return t if BF1 and BF2 are not the same set of files or buffers."
```

Comment:
This should probably say "Return non-nil if...", because it shouldn't matter whether it's `t` or another non-nil value.

Also, it would be good to explain what BF1 and BF2 are (checkdoc will complain otherwise, at least).